### PR TITLE
[FIX] BroadcastJob activeDeadlineSeconds did not take effect

### PR DESCRIPTION
**Pull request by ls-2018**
- https://github.com/openkruise/kruise/issues/1409

A description of the specific problem is in issue

### CODE REVIEW

File: pkg/util/handler/enqueue.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+	appsalphav1 "github.com/openkruise/kruise/apis/apps/v1alpha1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+type empty struct{}
```
Comment (zmberg) [inline pkg/util/handler/enqueue.go:30]: Why must this official function be redefined

**Comment by ls-2018** [comment]
I simplified the code a little bit.


Reconcile all of the Reconcile function can be handled because AdvancedCronJob, BroadcastJob and Job events occur. However, in the case I submitted, I watched three kinds of resources respectively and found that no incident happened. Of course, this probability may be low, but it is not impossible

**Comment by zmberg** [comment]
What I'm more interested in understanding is how this patch related to the issue? Bug?

**Comment by ls-2018** [comment]
I think it was a bug. 
Pulling the large image exceeded the expected death time, but the pod could not be killed because no event was watched.

**Comment by zmberg** [comment]
@ls-2018 You are right, and i think you can refer the implement, https://github.com/openkruise/rollouts/blob/master/pkg/controller/rollout/rollout_controller.go#L167

**Comment by ls-2018** [comment]
I think I have found the root cause. It is not that the ActiveDeadlineSeconds have not been reconciled after expiration, but that there is no judgment on whether the pod should be removed or not.
@zmberg

**Comment by ls-2018** [comment]
@zmberg  Can you look at this request again please? Thanks.

**Pull request closed without merging**